### PR TITLE
Removing SimpleMock test from pandas.util.testing

### LINKED ISSUE #21475: CLN: remove SimpleMock from pandas.util.testing
not used anywhere

### PR DESCRIPTION
**Pull request by uds5501**
Removing SimpleMock test

- [x] closes #21475
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by TomAugspurger** [comment]
restarted the HTTP failure.

File: pandas/util/testing.py [hunk 1]
```diff
@@ -2263,37 +2263,6 @@ def wrapper(*args, **kwargs):
 with_connectivity_check = network
 
 
-class SimpleMock(object):
```
Comment (TomAugspurger) [inline pandas/util/testing.py]: Looks like SimpleMock is used in `stdin_encoding` below. Can you see if `stdin_encoding` is used anywhere, and maybe remove it too?

Comment (uds5501) [reply pandas/util/testing.py]: @TomAugspurger Sure, will do

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by jreback** [comment]
thanks @uds5501

**Pull request merged**